### PR TITLE
`DPNPC_ptr_adapter` integration into random multinomial

### PR DESCRIPTION
**Pull request by LukichevaPolina**

### CODE REVIEW

File: dpnp/backend/kernels/dpnp_krnl_random.cpp [hunk 1]
```diff
@@ -456,6 +457,9 @@ void dpnp_rng_multinomial_c(
         }
         else
         {
+            DPNPC_ptr_adapter<double> p_vector_ptr(p_vector, p_vector_size);
+            double* p_vector1 = p_vector_ptr.get_ptr();
```
Comment (densmirn) [inline dpnp/backend/kernels/dpnp_krnl_random.cpp:461]: `p_vector1` is not used at all.

File: dpnp/backend/kernels/dpnp_krnl_random.cpp [hunk 2]
```diff
@@ -435,7 +435,8 @@ void dpnp_rng_multinomial_c(
     }
     else
     {
-        std::int32_t* result1 = reinterpret_cast<std::int32_t*>(result);
+        DPNPC_ptr_adapter<std::int32_t> result_ptr(result, size, true, true);
```
Comment (densmirn) [inline dpnp/backend/kernels/dpnp_krnl_random.cpp:438]: Parameter `target_no_sycl` is `true`. Do you think it's correct value for the parameter?

File: dpnp/backend/kernels/dpnp_krnl_random.cpp [hunk 3]
```diff
@@ -456,6 +457,9 @@ void dpnp_rng_multinomial_c(
         }
         else
         {
+            DPNPC_ptr_adapter<double> p_vector_ptr(p_vector, p_vector_size);
+            double* p_vector1 = p_vector_ptr.get_ptr();
+            
             int errcode = viRngMultinomial(
                 VSL_RNG_METHOD_MULTINOMIAL_MULTPOISSON, get_rng_stream(), n, result1, ntrial, p_vector_size, p_vector);
```
Comment (samir-nasibli) [inline dpnp/backend/kernels/dpnp_krnl_random.cpp:464]:
``` suggestion
                VSL_RNG_METHOD_MULTINOMIAL_MULTPOISSON, get_rng_stream(), n, result1, ntrial, p_vector_size, p_vector1);
```
```

File: dpnp/backend/kernels/dpnp_krnl_random.cpp (hunk 2 above)
Comment (LukichevaPolina) [reply dpnp/backend/kernels/dpnp_krnl_random.cpp:438]: Parameter `target_no_sycl` with `false` value for result_ptr leads to _Segmentation fault_ when calling `viRngMultinomial()`.

**Review by densmirn — APPROVED** [review]

**Review by densmirn — APPROVED** [review]

**Review by Alexander-Makaryev — APPROVED** [review]
LGTM

**Pull request merged**